pkg/cdi/cache.go

```diff
@@ -17,18 +17,31 @@
 package cdi
 
 import (
+	"io/fs"
+	"os"
 	"path/filepath"
 	"sort"
 	"strings"
 	"sync"
 
+	stderr "errors"
+
 	cdi "github.com/container-orchestrated-devices/container-device-interface/specs-go"
 	"github.com/fsnotify/fsnotify"
 	"github.com/hashicorp/go-multierror"
 	oci "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/pkg/errors"
 )
 
+const (
+	// reservationDir is the subdirectory for transient Spec reservations.
+	reservationDir = ".transient"
+	// transientPrefix is the base prefix used to transient Spec files.
+	transientPrefix = "transient."
+	// transientExt is the type/extension for transient Spec files.
+	transientExt = ".json"
```

Comment:
+1 for using YAML. JSON is good as a transport protocol in my opinion. YAML is more suitable for storing configuration info and extensively used in k8s for this purpose.

Reply:
I'd also prefer YAML. I went with JSON because I thought there is a strong preference for it (the original CDI implementation did not support YAML at all). Is there a need to let the user choose the format for these generated files or can we always go with YAML here ?

Reply:
It would be better to let the user choose the format, but default should be YAML I think.

Reply:
YAML is also preferable considering @klueska's opinion that most of plugins would end up using static CDI files. His NVIDIA GPU DRA plugin only uses static files.

Reply:
Changed the default encoding to YAML.